test/functional/operation_example.test.js

```diff
@@ -2024,7 +2023,7 @@ describe('Operation Examples', function() {
             o.out = { replace: 'replacethiscollection' };
 
             collection.mapReduce(map, reduce, o, function(err, outCollection) {
-              test.equal(null, err);
+              expect(err).to.be.null;
```

Comment:
`expect(err).to.not.exist` not `to.be.null` (I think this happens in a few other places)

Reply:
https://www.chaijs.com/api/bdd/#method_exist
Chai recommends to be specific about the assertion which is why I went this way, but If you think its fine for our case to go with general existence I have no further qualms

Reply:
we commonly use `to.not.exist` throughout the code base, primarily because we want to eventually be able to support emitting `undefined` rather than `null` when an error doesn't exist (reduce allocations). I think eventually we want to use a custom chai assertion that will print the error out if it exists, so if we use the same `to.not.exist` it will likely make a search+replace much easier.